python/ray/serve/api.py

```diff
@@ -299,8 +278,6 @@ class MyDeployment:
             this deployment. Defaults to 1.
         autoscaling_config: Parameters to configure autoscaling behavior. If this
             is set, `num_replicas` cannot be set.
-        route_prefix: [DEPRECATED] Route prefix should be set per-application
```

Comment:
If we are removing `route_prefix` from @serve.deployment, should we also remove from `.options()`? Also, I think there are still a few stray tests that use `route_prefix`.

Reply:
Yes we should... that might be more involved. Let me revert the route_prefix part here and can you make a separate PR to remove those?

Reply:
Yeah, I can handle removing route  prefix.